.github/workflows/ci.yml

```diff
@@ -0,0 +1,146 @@
+          coverage: none
+
+      - name: Install dependencies
+        run: composer install --no-ansi --no-interaction --no-progress
+
+      - name: Run linter
+        run: composer run cs
+
+  tests:
+    name: Tests
+
+    runs-on: ${{ matrix.os }}
+
+    env:
+      PHP_EXTENSIONS: dom, json, libxml, mbstring, pdo_sqlite, soap, xml, xmlwriter
+      SHELL_INTERACTIVE: true
+
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+
+        php-version:
+          - "7.4"
+
+        php-ini-values:
+          - assert.exception=1, zend.assertions=1
+
+        dependencies:
+          - locked
+        
+        codecov:
+          - false
+
+        include:
+          - os: ubuntu-latest
+            php-version: "5.5.38"
+            
+          - os: ubuntu-latest
+            php-version: "5.6"
+            dependencies: highest  
+            
+          - os: ubuntu-latest
+            php-version: "7.0"
+            dependencies: highest
+
+          - os: ubuntu-latest
+            php-version: "7.1"
+            dependencies: highest
+              
+          - os: ubuntu-latest
+            php-version: "7.2"
+            dependencies: highest
+            
+          - os: ubuntu-latest
+            php-version: "7.3"
+            dependencies: highest  
+
+          - os: ubuntu-latest
+            php-version: "7.4"
+            dependencies: highest
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          coverage: pcov
+          extensions: ${{ env.PHP_EXTENSIONS }}
+          ini-values: ${{ matrix.php-ini-values }}
+
+      - name: Determine composer cache directory on Linux
+        if: matrix.os == 'ubuntu-latest'
+        run: echo "COMPOSER_CACHE_DIR=$(composer config cache-dir)" >> $GITHUB_ENV
+
+      - name: Determine composer cache directory on Windows
+        if: matrix.os == 'windows-latest'
+        run: Add-Content -Path $ENV:GITHUB_ENV -Value "COMPOSER_CACHE_DIR=~\AppData\Local\Composer"
+
+      - name: Cache dependencies installed with composer
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.COMPOSER_CACHE_DIR }}
+          key: php${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-
+
+      - name: Make sure composer.json is valid before we start modifyig it
+        run: composer validate
+        
+      - name: Clear platform php configuration in case we need to update phpunit
+        run: composer config --unset platform.php
+
+      - name: Update phpunit if dependencies are locked in case phpunit version in lock file is not compatible
+        if: matrix.dependencies == 'locked'
```

Comment:
Exactly one of the 'locked', 'highest' or 'lowest' sections should run, but in your failing php 5.5 test none are running. I don't know why this is.

Reply:
n.b. it is expected that if one test fails, GitHub actions will cancel the others.

Reply:
There's a setting to disable this behavior if you want to.